tests/text/test_parsing.py

```diff
@@ -171,107 +174,156 @@ def test_partial_match_preprocessing():
 NAME: /[^\W\d]\w*/
 
+
     """,
         parser="lalr",
         postlex=PartialPythonIndenter(),
     )
 
-    terminals_to_states = terminals_to_lalr_states(pyparser)
-    symbol_names_and_fsms = terminals_to_fsms(pyparser)
+    terminals_to_states = terminals_to_lalr_states(lp)
+    symbol_names_and_fsms = terminals_to_fsms(lp)
 
     term_type = "DEF"
-    def_fsm = symbol_names_and_fsms[term_type]
+    term_fsm = symbol_names_and_fsms[term_type]
 
-    # TODO FIXME: This is broken, and it's a bug in `lark`'s Python grammar!
+    # TODO FIXME: This is broken, and it's a bug in `lark`'s Python grammar?
     # ptoken = "defx"
 
     ptoken = "ef foo"
-    pmatch = find_partial_matches(def_fsm, ptoken)
-    first_pmatch = next(pm for pm in pmatch if pm[0] is not None)
+    pmatches = find_partial_matches(term_fsm, ptoken)
+    first_pmatch = next(pm for pm in pmatches if pm[0] is not None)
     (parser_state,) = create_pmatch_parser_states(
-        pyparser, terminals_to_states, term_type, ptoken, first_pmatch
+        lp, terminals_to_states, term_type, ptoken, first_pmatch
     )
     new_parser_state, expected_next_tokens = parse_to_end(parser_state)
     assert expected_next_tokens == {"NAME"}
 
     ptoken = "ef foo():"
-    pmatch = find_partial_matches(def_fsm, ptoken)
-    first_pmatch = next(pm for pm in pmatch if pm[0] is not None)
+    pmatches = find_partial_matches(term_fsm, ptoken)
+    first_pmatch = next(pm for pm in pmatches if pm[0] is not None)
     (parser_state,) = create_pmatch_parser_states(
-        pyparser, terminals_to_states, term_type, ptoken, first_pmatch
+        lp, terminals_to_states, term_type, ptoken, first_pmatch
     )
     new_parser_state, expected_next_tokens = parse_to_end(parser_state)
     assert not expected_next_tokens
 
     ptoken = "ef ("
-    pmatch = find_partial_matches(def_fsm, ptoken)
-    first_pmatch = next(pm for pm in pmatch if pm[0] is not None)
+    pmatches = find_partial_matches(term_fsm, ptoken)
+    first_pmatch = next(pm for pm in pmatches if pm[0] is not None)
     (parser_state,) = create_pmatch_parser_states(
-        pyparser, terminals_to_states, term_type, ptoken, first_pmatch
+        lp, terminals_to_states, term_type, ptoken, first_pmatch
     )
     with pytest.raises(UnexpectedToken):
         parse_to_end(parser_state)
+
+
+def test_map_partial_states_to_vocab_regex():
+    regex_string = r"(([0-9]+)?([.]([0-9]*)?)?|[.][0-9]+)"
+    regex_pattern = interegular.parse_pattern(regex_string)
+    regex_fsm = regex_pattern.simplify().to_fsm()
+
+    vocabulary = ["1.", "2", "3.", ".", ".80", "42", "1a", " ", "0", "a", "b", "$"]
+
+    # We want the vocabulary strings to entirely match the regex--not just the
+    # prefixes of the vocabulary strings
+    def partial_match_filter(string, end_idx, state_seq):
+        if end_idx is not None and end_idx < len(string) - 1:
+            return False
+        return True
+
+    pstate_to_vocab = map_partial_states_to_vocab(
```

Comment:
We should add function that maps the states of masks over the vocabulary, but this can be done in a later PR when needed, #166 for instance.